### PR TITLE
[#116] initial implementation of type hints

### PR DESCRIPTION
**Pull request by pkdash**
This is an initial implementation of adding type hints that includes type hints just for one function of `nwis` module. If this looks good, I can implement the same for the rest of the functions of the `nwis `module.

### CODE REVIEW

File: dataretrieval/nwis.py [hunk 1]
```diff
@@ -270,28 +272,31 @@ def _discharge_measurements(ssl_check=True, **kwargs):
     return _read_rdb(response.text), NWIS_Metadata(response, **kwargs)
 
 
-def get_discharge_peaks(sites=None, start=None, end=None,
-                        multi_index=True, ssl_check=True, **kwargs):
+def get_discharge_peaks(sites: Optional[Union[List[str], str]] = None,
+                        start: Optional[str] = None, end: Optional[str] = None,
+                        multi_index: bool = True,
+                        ssl_check: bool = True, **kwargs) -> tuple[pd.DataFrame, BaseMetadata]:
```
Comment (thodson-usgs) [inline dataretrieval/nwis.py]:
```suggestion
                        ssl_check: bool = True, **kwargs) -> Tuple[pd.DataFrame, BaseMetadata]:
```

File: dataretrieval/nwis.py [hunk 2]
```diff
@@ -12,13 +12,15 @@
 
 """
 
+import re
 import warnings
-import pandas as pd
 from io import StringIO
-import re
+from typing import List, Optional, Union
```
Comment (thodson-usgs) [inline dataretrieval/nwis.py]:
```suggestion
from typing import List, Optional, Union, Tuple
```

**Review by thodson-usgs — COMMENTED** [review]
Excellent @pkdash!
I noticed the 3.8 tests were failing. I suspect `tuple` is only supported for 3.9+.
Please make this change and all tests should pass.

File: dataretrieval/nwis.py (hunk 1 above)
Comment (pkdash) [reply dataretrieval/nwis.py]: @thodson-usgs Thanks for catching this type hint issue.

**Pull request merged**